### DISCUSSION TITLE
[Deploy] Smart deploy comparison not working - Unable to get cronjobs from environment

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#302] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v10.8.0

### Description

Hi guys,
I am getting an error on the Deploy page, I am not able to see the "microservices to be deployed" table:
<img width="1117" alt="Screenshot 2023-09-01 at 10 12 23" src="https://github.com/mia-platform/community/assets/28152586/8a4db45f-5026-460a-998d-8f1012cbee02">

Following are the request details:
```
curl 'https://{{CONSOLE_URL}}/api/deploy/projects/{{PROJECT_ID}}/compare?toRef=newBranch&fromRef=actualBranch&fromEnvironment=DEV'
```

The response is the following:
```
{
    "statusCode": 500,
    "error": "Internal Server Error",
    "message": "unable to get cronjobs from environment"
}
```

How can I fix it?

### Actual Outcome

The comparison table is not shown.

### Expected Outcome

I see the comparison table.

**Comment by fredmaggiowski** [comment]
Hi, this is most likely a problem with the authorizations provided to the service account on the cluster you are using.

Please refer to the following doc to check what is missing: https://docs.mia-platform.eu/docs/paas/runtime#permissions

**Reply by frengooo** [reply]
Hi @fredmaggiowski, thanks for your reply.

Ok, but why it's happening now? We didn't modify service account permissions and it always worked before.

**Reply by frengooo** [reply]
Hi @fredmaggiowski,
the service account has requested permissions.

The console version is v10.8, could it be related to the `v1beta` cronjob's deprecation?
In that case, how can we fix it?

**Reply by fredmaggiowski** [reply]
OK thank you for the information, what is the cluster version?

CronJob v1beta1 has been dismissed with v11.0.0 so they are still supported by your Platform version

**Reply by fredmaggiowski** [reply]
@silversoul93 can you tell me the k8s cluster version please?

**Reply by fredmaggiowski** [reply]
If you are using the multitenant cluster this may be due to the fact that a k8s version > 1.25 is being used and as such is incompatible with your CronJobs using apiVersion `batch/v1beta1`

You should update your CronJob yaml file to `batch/v1`

**Reply by frengooo** [reply]
Hi @fredmaggiowski, I am using the multitenant noprod cluster, but I don't have any cronjob in my project configurations or in the related namespaces.

**Reply by fredmaggiowski** [reply, marked as answer]
This might still be a problem with the available APIs, the Console is trying to get CronJobs from `v1beta1` but these api version is unknown to the cluster.

I suggest you to open a SR to address this issue with the PaaS infrastructure team.

**Reply by frengooo** [reply]
Thanks @fredmaggiowski
